tests/sentry/utils/test_redis.py

```diff
@@ -0,0 +1,84 @@
+import functools
+
+import pytest
+
+from sentry.exceptions import InvalidConfiguration
+from sentry.testutils.cases import TestCase
+from sentry.utils.redis import (
+    ClusterManager, _shared_pool, get_cluster_from_options
+)
+
+make_manager = functools.partial(
+    ClusterManager,
+    {
```

Comment:
This is a bit of a hack — it relies on `OptionManager.get` and `dict.get` behaving the same. This does isolate state much better, though.